code/modules/research/design_database_analyzer.dm

```diff
@@ -0,0 +1,143 @@
+/obj/machinery/destructive_analyzer
+	name = "destructive analyzer"
+	desc = "Accessed by a connected design database console, it destroys and analyzes items and materials for later reproduction."
+	icon = 'icons/obj/machines/destructive_analyzer.dmi'
+	icon_state = "d_analyzer"
+	idle_power_usage = 30
+	active_power_usage = 2500
+	construct_state = /decl/machine_construction/default/panel_closed
+	density = TRUE
+	anchored = TRUE
+
+	var/local_network_tag
+	var/busy = FALSE
+	var/obj/item/loaded_item = null
+	var/material_return_modifier = 0
+	var/list/cached_materials = list()
+
+/obj/machinery/destructive_analyzer/examine(mob/user)
+	. = ..()
+	
+/obj/machinery/destructive_analyzer/Initialize()
+	. = ..()
+	set_extension(src, /datum/extension/local_network_member)
+	if(local_network_tag)
+		var/datum/extension/local_network_member/lanm = get_extension(src, /datum/extension/local_network_member)
+		lanm.set_tag(null, local_network_tag)
+
+/obj/machinery/destructive_analyzer/RefreshParts()
+	var/T = 0
+	for(var/obj/item/stock_parts/S in src)
+		T += S.rating
+	material_return_modifier = min(T * 0.05, 1)
+	..()
+
+/obj/machinery/destructive_analyzer/on_update_icon()
+	if(panel_open)
+		icon_state = "d_analyzer_t"
+	else if(loaded_item)
+		icon_state = "d_analyzer_l"
+	else
+		icon_state = "d_analyzer"
+
+/obj/machinery/destructive_analyzer/components_are_accessible(path)
+	return !busy && ..()
+
+/obj/machinery/destructive_analyzer/cannot_transition_to(state_path)
+	if(busy)
+		return SPAN_WARNING("\The [src] is busy. Please wait for completion of previous operation.")
+	if(loaded_item)
+		return SPAN_WARNING("There is something already loaded into \the [src]. You must remove it first.")
+	return ..()
+
+/obj/machinery/destructive_analyzer/attack_hand(mob/user)
+
+	if(loaded_item)
+		loaded_item.dropInto(user.loc)
+		user.put_in_hands(loaded_item)
+		loaded_item = null
+		update_icon()
+		return TRUE
+
+	var/list/dump_matter
+	for(var/mat in cached_materials)
+		var/amt = Floor(cached_materials[mat]/SHEET_MATERIAL_AMOUNT)
+		if(amt > 0)
+			LAZYSET(dump_matter, mat, amt)
+	if(length(dump_matter))
+		visible_message("\The [user] unloads \the [src]'s material hopper.")
+		for(var/mat in dump_matter)
+			var/material/M = SSmaterials.get_material_datum(mat)
+			M.place_sheet(loc, dump_matter[mat])
+			cached_materials[mat] -= dump_matter[mat] * SHEET_MATERIAL_AMOUNT
+			if(cached_materials[mat] <= 0)
+				cached_materials -= mat
+		return TRUE
+
+	. = ..()
```

Comment:
Use `physical_attack_hand` and `interface_interact` instead for machines.

Reply:
Swapped in `physical_attack_hand`, not sure where you want `interface_interact` used as I used it for the other machines.